File: 4/ts/fruits.ts
/**
 * This is the standard index.ts starting point.
 *
 * author Dominic M.
 * version 1.0
 * since 2024-10-16
 */
import { MrCoxallStack } from './MrCoxallStack.ts'

// Use MrCoxallStack class
const fruitStack = new MrCoxallStack()
fruitStack.pushString('red')
fruitStack.pushString('green')
fruitStack.pushString('blue')
console.log(`Fruit items: ${fruitStack.getStack}`)
console.log(`The size of the stack is: ${fruitStack.getSize}`)
console.log(`Is the stack empty? ${fruitStack.getEmpty.toString()}`)
console.log(`Popped fruit item: ${fruitStack.popItem()}`)
console.log(`Fruit items: ${fruitStack.getStack}`)
console.log(`The size of the stack is: ${fruitStack.getSize}`)
console.log(`Is the stack empty? ${fruitStack.getEmpty.toString()}`)
console.log(`Top fruit is: ${fruitStack.peekItem}`)
console.log(`The size of the stack is: ${fruitStack.getSize}`)
console.log("Emptying stack...")
fruitStack.clearStack()
console.log(`Fruit items: ${fruitStack.getStack}`)
console.log(`The size of the stack is: ${fruitStack.getSize}`)
console.log(`Is the stack empty? ${fruitStack.getEmpty.toString()}`)
console.log('\nDone.')

File: 4/java/Fruits.java
/*
 * The Starting point for the script.
 *
 * @author Dominic M.
 * @since 2024-10-22
 * @version 1.0
 */

/**
 * This is the Fruits class.
 */
final class Fruits {

    /**
     * Prevent instantiation.
     *
     * @throws IllegalStateException if this is ever called
     *
     */
    private Fruits() {
        throw new IllegalStateException("Cannot be instantiated");
    }

    /**
     * This is the main function.
     *
     * @param args no args used
     */
    public static void main(final String[] args) {

        // Use MrCoxallStack class
        final MrCoxallStack fruitStack = new MrCoxallStack();
        fruitStack.pushString("apple");
        fruitStack.pushString("banana");
        fruitStack.pushString("cherry");
        System.out.println("Fruit items: " + fruitStack.getStack());
        System.out.println("The stack's size is: " + fruitStack.getSize());
        System.out.println(
            "Is the stack empty? " + String.valueOf(fruitStack.getEmpty())
        );
        System.out.println("Popped fruit item: " + fruitStack.popItem());
        System.out.println("Fruit items: " + fruitStack.getStack());
        System.out.println("The stack's size is: " + fruitStack.getSize());
        System.out.println(
            "Is the stack empty? " + String.valueOf(fruitStack.getEmpty())
        );
        System.out.println("Top fruit is: " + fruitStack.peekItem());
        System.out.println("The stack's size is: " + fruitStack.getSize());
        System.out.println("Emptying stack...");
        fruitStack.clearStack();
        System.out.println("Fruit items: " + fruitStack.getStack());
        System.out.println("The stack's size is: " + fruitStack.getSize());
        System.out.println(
            "Is the stack empty? " + String.valueOf(fruitStack.getEmpty())
        );

        System.out.println("\nDone.");
    }
}
